common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -316,25 +317,31 @@ public AcquireTokenResult acquireTokenSilent(
         // subsequent CacheRecords represent other profiles (projections) of this principal in
         // other tenants. Those tokens will be 'sparse', meaning that their AT/RT will not be loaded
         final ICacheRecord fullCacheRecord = cacheRecords.get(0);
+        if (fullCacheRecord.getAccessToken().refreshOnIsActive()) {
+            Logger.info(
+                    TAG,
+                    "RefreshOn is active. This will extend your token usage in the rare case servers are not available."
+            );
+        }
 
-        if (accessTokenIsNull(fullCacheRecord)
+        if(fullCacheRecord.getAccessToken().shouldRefresh()){
+            setAcquireTokenResult(acquireTokenSilentResult, parametersWithScopes, cacheRecords);
+            final RefreshOnCommand refreshOnCommand = new RefreshOnCommand(parameters, this, "LocalMSALControllerMockPubId");
```

Comment:
We should pick something better than `LocalMSALControllerMockPubId`

Reply:
It was a placeholder actually. A question I had is if we need to have the publicApiId for commands with void return types? It's hard to obtain this String while in the Controllers.